homeassistant/components/homekit_controller/__init__.py

```diff
@@ -0,0 +1,215 @@
+                if ctype != 'serial-number':
+                    continue
+                return characteristic['value']
+        return None
+
+    def accessory_setup(self):
+        import homekit
+        self.controllerkey, self.accessorykey = \
+            homekit.get_session_keys(self.conn, self.pairing_data)
+        self.securecon = homekit.SecureHttp(self.conn.sock,
+                                            self.accessorykey,
+                                            self.controllerkey)
+        response = self.securecon.get('/accessories')
+        data = json.loads(response.read().decode())
+        for accessory in data['accessories']:
+            serial = self.get_serial(accessory)
+            if serial in self.hass.data[KNOWN_ACCESSORIES]:
+                continue
+            self.hass.data[KNOWN_ACCESSORIES][serial] = self
+            aid = accessory['aid']
+            for service in accessory['services']:
+                service_info = {'serial': serial,
+                                'aid': aid,
+                                'iid': service['iid']}
+                devtype = homekit.ServicesTypes.get_short(service['type'])
+                _LOGGER.debug("Found %s", devtype)
+                component = HOMEKIT_ACCESSORY_DISPATCH.get(devtype, None)
+                if component is not None:
+                    discovery.load_platform(self.hass, component, DOMAIN,
+                                            service_info, self.config)
+
+    def device_configuration_callback(self, callback_data):
+        """Handle initial pairing"""
+        import homekit
+        pairing_id = str(uuid.uuid4())
+        code = callback_data.get('code').strip()
+        self.pairing_data = homekit.perform_pair_setup(
+            self.conn, code, pairing_id)
+        if self.pairing_data is not None:
+            homekit.save_pairing(self.pairing_file, self.pairing_data)
+            self.accessory_setup()
+        else:
+            error_msg = "Unable to pair, please try again"
+            _configurator = self.hass.data[DOMAIN+self.hkid]
+            self.configurator.notify_errors(_configurator, error_msg)
+
+    def configure(self):
+        description = """Please enter the HomeKit code for your {}""".format(self.model)
+        self.hass.data[DOMAIN+self.hkid] = \
+            self.configurator.request_config(self.model,
+                                             self.device_configuration_callback,
+                                             description=description,
+                                             submit_caption="submit",
+                                             fields=[{'id': 'code',
+                                                      'name': 'HomeKit code',
+                                                      'type': 'string'}])
+
+
+class HomeKitEntity(Entity):
+    """Representation of a Home Assistant HomeKit device."""
+
+    def __init__(self, hass, devinfo):
```

Comment:
I would prefer if you pass in only the `accessory`. That makes it more clear what is needed.